documentation/src/docs/security/generating-tls-artifacts.md

```diff
@@ -0,0 +1,324 @@
+  $ keytool -keystore server01.keystore.jks -alias server01 -genkey
+  ```
+* A weak password `changeit` is used everywhere in this document for easier reading. Be sure to replace it with 
+  a strong and separate password for each file.
+
+## Setting up a Certificate Authority (CA)
+
+If you are planning to use a public CA or an existing private/internal CA, you may skip 
+this part altogether and go directly to the [next section](#generate-tls-certificates-and-other-tls-artifacts-for-the-server).
+
+Here, we'll create a CA in the form of a public/private key pair and a self-signed certificate.
+Later, we'll use this CA certificate/key bundle to sign the servers' certificates.
+
+1. Generate a CA certificate and public/private key pair. 
+
+   ```bash
+   $ openssl req -new -x509 -keyout <ca-private-key-file-path> -out <ca-cert-file-path> -days <validity-in-days> \
+            -subj "<distinguished_name>" \
+            -passout pass:<ca-private-key-password>
+   ```
+
+   Example:
+   ```bash
+   $ openssl req -new -x509 -keyout ca-key.key -out ca-cert.crt -days 365 \
+            -subj "/C=US/ST=Washington/L=Seattle/O=Pravega/OU=CA/CN=Pravega-Stack-CA" \
+            -passout pass:changeit
+   ```
+  
+   The command above will generate the following two PEM-encoded files: 
+   * A file containing the encrypted private key.
+   * A file containing the CA certificate.
+   
+2. Optionally, create a Java truststore containing the CA's certificate. This may be used by client applications if 
+   they configure the truststore using the `javax.net.ssl.trustStore` Java option. 
+   
+   ```bash
+   $ keytool -keystore <java-truststore-file-path> -noprompt \
+             -alias <trusted-cert-entry-alias> \
+             -import -file <ca-cert-file-path> \
+             -storepass <java-truststore-file-password>
+   ```
+   
+   Example:
+   ```bash
+   $ keytool -keystore client.truststore.jks -noprompt \
+             -alias caroot \ 
+             -import -file ca-cert.crt \
+             -storepass changeit
+
+   # Optionally, list the truststore's contents and inspect the output to verify everything is in order. The output 
+   # should show a single entry with alias name `caroot` and entry type `trustedCertEntry`.
+   $ keytool -list -v -keystore client.truststore.jks -storepass changeit
+   ```
+
+At this point, the following CA and truststore artifacts should be ready:
+
+| File | Description | Example command for inspecting the file's Contents |
+|:-----:|:--------|:--------------|
+| `ca-cert.crt` | PEM-encoded X.509 certificate of the CA | `$ openssl x509 -in ca-cert.crt -text -noout` |
+| `ca-key.key` | PEM-encoded file containing the CA's encrypted private key  | `$ openssl pkcs8 -inform PEM -in ca-key.key -topk8` |
+| `client.truststore.jks` | A password-protected truststore file containing the CA's certificate | `$ keytool -list -v -keystore client.truststore.jks -storepass changeit` |
```

Comment:
Both `ca-key.key` and `client.truststore.jks` are password-protected or encrypted. Consider using the same phrasing for both entries.

Reply:
There is a slight difference though: the JKS is a password protected file, while the key file contains the encrypted key in a PEM file. The key file itself is not encrypted. So, I have left that part of the text as-is. What do you say?

Reply:
Yes, I see what you mean. An ASN.1 attribute in the PEM file is encrypted, whereas the entire truststore is encrypted? Your wording is fine!